docs/designs/2730-visibility.md

```diff
@@ -0,0 +1,293 @@
+# Improved Bootstrap Visibility
+
+A proposal for updating our console output on cluster creation to make it
+easier for end users to understand what is happening.
+
+* Proposal:
+  [https://github.com/vmware-tanzu/community-edition/issues/2730](https://github.com/vmware-tanzu/community-edition/issues/2730)
+
+## Summary
+
+One of the most common support requests we've received with Tanzu Community
+Edition has been around failures bootstrapping new clusters. To make the
+bootstrap process less confusing, and easier to troubleshoot when things go
+wrong, this proposal is to update the bootstrap output for management and
+workload cluster deployment to be more concise and targeted about what
+information is given, with better context about the steps and what actions are
+being performed for each one.
+
+## Motivation
+
+Often, the cause of cluster deployment failure ends up being fairly obvious
+once the proper logs are found or statuses are checked. The hard part is
+usually understanding what has happened and where to look to find this
+information.
+
+Whether the cluster deployment is successful or fails, it can also be confusing
+to understand what is happening during deployment. This lack of visibility can
+be a challenge to understand where things failed, but it can also be confusing
+during a successful deployment. Especially for new users, this lack of
+visibility can introduce uncertainty that decrease confidence in Tanzu
+Community Edition.
+
+Simplifying the output of our deployment process, even though it provides less
+data along the way, should make it more clear what is happening and help users
+understand where in the deployment process we are. That gives them some
+confidence that things are progressing, and help them understand what is
+happening during that process.
+
+Better output on failure should lead the user to finding root cause and faster
+resolution of blocking issues.
+
+### Goals
+
+* Minimize what gets emitted to the console during cluster deployment to only
+  what the user needs to see
+* Allow more details only if requested when the user requests verbose output
+* Group output into the large grain steps of the process
+* Within each step, only emit the details related to that step that may be
+  useful for the end user to know
+* In case of failure, perform some basic triaging to provide the user with
+  enough details to help them correct the situation
+* Provide clear and direct next steps for them to perform
+
+The hope is to move from the unformatted and sometimes messy output we have
+today:
+
+![current](imgs-2730/current.png)
+
+to a more structured and visually easier to parse format similar to the `kind`
+output:
+
+![kind](imgs-2730/kind.png)
+
+or the planned output for the proposed `unmanaged-cluster` command:
```

Comment:
The _current_ output of `unamanged-cluster`? 😬

Reply:
Updated!